knip.jsonc

```diff
@@ -1,7 +1,9 @@
 {
 	"$schema": "https://unpkg.com/knip@next/schema.json",
-	"entry": "src/index.ts!",
-	"project": "src/**/*.ts!",
-	// TODO: https://github.com/JoshuaKGoldberg/ts-api-utils/issues/45
-	"ignore": ["src/nodes/typeGuards/!(*.test).ts"]
+	"entry": ["src/**/index.ts!", "src/**/*.test.ts"],
+	"project": [
+		"src/!(test)/**/!(*.test).ts!",
+		"src/!(*.test).ts!",
+		"src/**/*.ts"
+	]
```

Comment:
🤔 I can't get `lint:knip:production` to fail on anything. I:
* Created a standalone file imported by a used file with an unused export
* Created a standalone file imported by a test with an unused export
* Created an unused file with an unused export
...nothing. How does `lint:knip:production` fail?

Reply:
https://github.com/JoshuaKGoldberg/ts-api-utils/pull/40#issuecomment-1429030453

Should work now with latest commit.